Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -198,17 +196,34 @@ public ITransformedNetwork getTransformedNetworkusingNetworkList(List<INetwork>
         // return new network with updated nodes and links
         return new TransformedNetwork(updatedNodes, links);
     }
-
-    private ITransformedNetwork getFinalTransformedNetwork(ITransformedNetwork transformedNetwork, String conceptId) {
-        // Filter the nodes with the concept id
+    /**
+     * Filter the nodes in the network using the concept id and the alternative id of the concept
+     * @param transformedNetwork 
+     * @param conceptId
+     * @param alternativeIdsForConcept
+     * @return ITransformedNetwork
+     */
+    private ITransformedNetwork getFinalTransformedNetwork(ITransformedNetwork transformedNetwork, String conceptId, List<String> alternativeIdsForConcept) {
+        // Filter the nodes with the concept id and alternative id of the concept
         // add all the statement ids to a set
         Set<String> statementIdSearchSet = new HashSet<String>();
         List<Node> searchedNodes = new ArrayList<Node>();
         for (Node node : transformedNetwork.getNodes().values()) {
+            // check if the node's concept id matches the searched concept id
             if (conceptId.equals(node.getConceptId())) {
                 searchedNodes.add(node);
                 statementIdSearchSet.addAll(node.getStatementIds());
             }
+            // check if the node's concept id matches the alternative id of the concept
```

Comment:
alternative ids should contain all ids (including the one that was searched for), so you can remove the previous if block.